### DISCUSSION TITLE
Changing `name` property of `useFieldArray` after it has been instantiated only works once

### DISCUSSION THREAD

**Discussion by nerdyman** [react-hook-form/react-hook-form#7725] [Q&A] [answered]
Hey! 

I've got a form that is split into different sections, the active section is controlled with a select outside of the hook form context, and when the section changes, a different set of components is rendered.

I'm using `useFieldArray` as the list of fields to render is in an array, however, I'm only able to get it to update one time when the section changes. For example, if the initial section is `one` and I change it to `two` using the select, it will update correctly. However, if I go back to `one` the rendered fields will still be from `two`. I'm not sure why this only works once.

I've made a basic repro here: https://codesandbox.io/s/interesting-wiles-04osh?file=/src/Form.tsx:157-173

And a quick recording of the issue:

https://user-images.githubusercontent.com/4908432/152211689-54b81b30-4f69-41fd-b75e-4b9b6ace386b.mp4

Any help would be awesome 🙌

**Comment by nerdyman** [comment, marked as answer]
I managed to get it working by also [applying the active section as the `key`](https://codesandbox.io/s/interesting-wiles-04osh?file=/src/Form.tsx:1661-1740) for the element that renders the field array.

![image](https://user-images.githubusercontent.com/4908432/152217941-bac36b28-33ed-4232-b40e-f5d44bfe338e.png)

Not sure if that's ideal though 🤔

**Reply by bluebill1049** [reply]
at the moment, we don't support `name` prop changing with `useFieldArray`.

**Reply by nerdyman** [reply]
Ah ok, thanks @bluebill1049! I'll mark the `key` method as the answer for now.
